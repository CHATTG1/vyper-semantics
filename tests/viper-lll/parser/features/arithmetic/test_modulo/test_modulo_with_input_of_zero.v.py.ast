%pgm(, ,
  %fdecl(%@public, foo, %param(a, %num) %param(b, %decimal), %decimal,
    %return(%binop(%, %var(a), %var(b))))
)
