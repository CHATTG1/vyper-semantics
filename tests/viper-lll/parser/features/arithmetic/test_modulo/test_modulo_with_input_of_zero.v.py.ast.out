$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3625405551 ) , $%seq ( $calldatacopy ( 320 , 4 , 64 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $pass , $%seq ( $mstore ( 0 , $clamp ( $mload ( 160 ) , $smod ( $mul ( $mload ( 320 ) , 10000000000 ) , $clamp_nonzero ( $mload ( 352 ) ) ) , $mload ( 128 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
